### PR TITLE
Refactor error handling

### PR DESCRIPTION
**Pull request by ptbrowne**
Instead of binding error handlers at import time, bind them
when BaseKonnector is built, it is cleaner. I don't remember
specifically why I started this refactor, but it was lingering
on my local branches.

### CODE REVIEW

File: packages/cozy-konnector-libs/src/libs/error.js [hunk 1]
```diff
@@ -1,19 +1,55 @@
...
+/**
+ * Attach event handlers to catch uncaught exceptions/rejections and signals.
+ * Log them as critical and exit the process accordingly.
+ * If the cleanup function has not been called, calling again the function
+ * is a no-op.
+ *
+ * @param  {Process} prcs - Process object, default to current process
+ * @return {Function} When called, removes the signal handlers
+ */
+const attachProcessEventHandlers = (prcs = process) => {
```
Comment (doubleface) [inline packages/cozy-konnector-libs/src/libs/error.js]: Why do you pass a prcs parameters when global process is used in event handlers anyway ?

File: packages/cozy-konnector-libs/src/libs/error.js [hunk 2]
```diff
@@ -1,19 +1,55 @@
...
+  }
+  attached = true
+  prcs.on('uncaughtException', handleUncaughtException)
+  prcs.on('unhandledRejection', handleUnhandledRejection)
+  prcs.on('SIGTERM', handleSigterm)
+  prcs.on('SIGINT', handleSigint)
+
+  return () => {
+    attached = false
+    prcs.off('uncaughtException', handleUncaughtException)
```
Comment (doubleface) [inline packages/cozy-konnector-libs/src/libs/error.js]: `EventEmitter.off` is marked as added in node v10 (https://nodejs.org/api/events.html#events_emitter_off_eventname_listener) and we still use node v8 in production.

Unit tests worked in node 8 anyway (no idea why) but could you use `removeListener` instead?

File: packages/cozy-konnector-libs/src/libs/error.js (hunk 1 above)
Comment (ptbrowne) [reply packages/cozy-konnector-libs/src/libs/error.js]: For tests.

File: packages/cozy-konnector-libs/src/libs/error.js (hunk 2 above)
Comment (ptbrowne) [reply packages/cozy-konnector-libs/src/libs/error.js]: Done, thanks :)

**Pull request merged**